src/symbolic/interface.jl

```diff
@@ -0,0 +1,78 @@
+export symbolize
+
+"""
+# Permutation symbolization
+
+    symbolize(x::AbstractVector{T}, est::SymbolicPermutation) where {T} → Vector{Int}
+    symbolize!(s, x::AbstractVector{T}, est::SymbolicPermutation) where {T} → Vector{Int}
+
+If `x` is a univariate time series, first `x` create a delay reconstruction of `x`
+using embedding lag `est.τ` and embedding dimension `est.m`, then symbolizing the resulting
+state vectors with [`encode_motif`](@ref).
+
+Optionally, the in-place `symbolize!` can be used to put symbols in a pre-allocated
+integer vector `s`, where `length(s) == length(x)-(est.m-1)*est.τ`.
+
+    symbolize(x::AbstractDataset{m, T}, est::SymbolicPermutation) where {m, T} → Vector{Int}
+    symbolize!(s, x::AbstractDataset{m, T}, est::SymbolicPermutation) where {m, T} → Vector{Int}
+
+If `x` is an `m`-dimensional dataset, then motif lengths are determined by the dimension of
+the input data, and `x` is symbolized by converting each `m`-dimensional
+state vector as a unique integer in the range ``1, 2, \\ldots, m-1``, using
+[`encode_motif`](@ref).
+
+Optionally, the in-place `symbolize!` can be used to put symbols in a pre-allocated
+integer vector `s`, where `length(s) == length(s)`.
```

Comment:
Wait, how can the lengths the same? Shouldn't `s` has less length `(d-1)*τ`?

Reply:
Each state vector `x` (a `D`-dimensional `Dataset`) gets transformed to a single integer symbol. The pre-allocated integer vector `s`  must fulfill `length(x) == length(s)`, so that `symbolize(x[i]) -> s[i]`.

Internally, a vector of length `D` is pre-allocated and re-used in the symbolization loop. 

However, the docs should of course read `length(s) == length(x)`, not `length(s) == length(s)`. Good catch!